### PR TITLE
Add username to order comments

### PR DESCRIPTION
**Pull request by seansan**
quick win for internal backend comments

also see: https://github.com/OpenMage/magento-lts/pull/329

### CODE REVIEW

**Comment by sreichel** [comment]
@seansan "quick win" ... LOL. :P

Next time please comment what you want to achive ... (please update)

File: app/code/core/Mage/Adminhtml/controllers/Sales/OrderController.php [hunk 1]
```diff
@@ -277,13 +277,23 @@ public function addCommentAction()
...
                 $visible = isset($data['is_visible_on_front']) ? $data['is_visible_on_front'] : false;
 
-                $order->addStatusHistoryComment($data['comment'], $data['status'])
+                $comment = $data['comment'];
+
+                $session = Mage::getSingleton('admin/session');
+                $username = $session->getUser()->getUsername();
+                
+                if (!$notify && !$visible) {
+        	        $comment .= (!issempty($username)) ? " ({$username})" : ' (system)';
```
Comment (spinsch) [inline app/code/core/Mage/Adminhtml/controllers/Sales/OrderController.php]: `issempty ` is not a valid php function.

Comment (sreichel) [inline app/code/core/Mage/Adminhtml/controllers/Sales/OrderController.php]: Please fix `isempty`

Comment (seansan) [reply app/code/core/Mage/Adminhtml/controllers/Sales/OrderController.php]: Changed to isset() myabe wiser than ($username)

Comment (tbaden) [reply app/code/core/Mage/Adminhtml/controllers/Sales/OrderController.php]: the variable is always set, even when false is returned. isn't empty() the right choice here?
https://www.virendrachandak.com/techtalk/php-isset-vs-empty-vs-is_null/

File: app/code/core/Mage/Adminhtml/controllers/Sales/OrderController.php [hunk 2]
```diff
@@ -277,13 +277,23 @@ public function addCommentAction()
                 $notify = isset($data['is_customer_notified']) ? $data['is_customer_notified'] : false;
                 $visible = isset($data['is_visible_on_front']) ? $data['is_visible_on_front'] : false;
 
-                $order->addStatusHistoryComment($data['comment'], $data['status'])
+                $comment = $data['comment'];
+
+                $session = Mage::getSingleton('admin/session');
+                $username = $session->getUser()->getUsername();
+                
+                if (!$notify && !$visible) {
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/controllers/Sales/OrderController.php]: Lines 282 and 283 should be after the `if` statement as they are only needed in `if`:

```
if (!$notify && !$visible) {
    $session = Mage::getSingleton('admin/session');
    $username = $session->getUser()->getUsername();
    $comment .= (!issempty($username)) ? " ({$username})" : ' (system)';
}

**Comment by sreichel** [comment]
Close in favor of #329.

(maybe someone puts on of this/both this in an extension)

**Pull request closed without merging**